src/Execution/Executor.php

```diff
@@ -946,6 +950,23 @@ protected function createResolveInfo(
         );
     }
 
+    /**
+     * @param ExecutionException $error
+     */
+    protected function handleError(ExecutionException $error)
+    {
+        $this->getErrorHandler()->handleError($error);
+        $this->context->addError($error);
+    }
+
+    /**
+     * @return ErrorHandlerInterface
+     */
+    protected function getErrorHandler(): ErrorHandlerInterface
+    {
+        return GraphQL::make(ErrorHandlerInterface::class);
```

Comment:
Wouldn't it be better to inject the error handler through the constructor?

Reply:
Hmm yes, you're right.

Reply:
Done.